plugins/dependencies/src/main/kotlin/dependencies/Kodein.kt

```diff
@@ -0,0 +1,68 @@
+@file:Suppress("PackageDirectoryMismatch", "SpellCheckingInspection", "unused")
+
+import org.gradle.api.Incubating
+import org.gradle.kotlin.dsl.IsNotADependency
+
+@Incubating
+object Kodein {
+
+
+    /**
+     * painless Kotlin dependency injection
+     *
+     * - [Official website here](https://kodein.org/di/)
+     * - GitHub page: [Kodein-Framework/Kodein-DI](https://github.com/Kodein-Framework/Kodein-DI)
+     * - [GitHub Releases here](https://github.com/Kodein-Framework/Kodein-DI/releases)
+     */
+    val di = DI
+
+    object DI : DependencyGroup(
+        group = "org.kodein.di",
+        rule = PrefixRule(versionKey = "version.kodein.di", prefix = "org.kodein.di:kodein-di")
+    ) {
+        val androidCore = module("kodein-di-framework-android-core")
+        val androidSupport = module("kodein-di-framework-android-support")
+        val androidx = module("kodein-di-framework-android-x")
+        val configurableJS = module("kodein-di-conf-js")
+        val configurableJvm = module("kodein-di-conf-jvm")
+        val js = module("kodein-di")
+        val jsr330 = module("kodein-di-jxinject-jvm")
+        val ktor = module("kodein-di-framework-ktor-server-jvm")
+        val tornadofx = module("kodein-di-framework-tornadofx-jvm")
+    }
+}
+
+
+open class DependencyGroup(
+    val group: String,
+    val rule: Rule? = null,
+    val usePlatformConstraints: Boolean = false
```

Comment:
We'd want to give users the choice, just like we do for Firebase BoM where you can access the `no-BoM` variant.

Reply:
Not a problem, see next commit